### PR TITLE
APIMF-2563: Metamodel modifications for Json Schema Draft 2019-09

### PR DESCRIPTION
**Pull request by tomsfernandez**
To be merged after release. Model version is still to be decided

### CODE REVIEW

**Review by antoniogarrote — DISMISSED** [review]
Would be good to make a pass at SHACL and double-check that there are not matching constraints we could map the new ones to.

**Comment by tomsfernandez** [comment]
Found the following SHACL mappings:
- DataArrangementModel / Contains -> sh:qualifiedValueShape (not changing it due to backwards compatibility)
- DataArrangementModel / MinContains -> sh:qualifiedMinCount
- DataArrangementModel / MaxContains -> sh:qualifiedMaxCount

File: amf-webapi/shared/src/main/scala/amf/plugins/domain/shapes/metamodel/NodeShapeModel.scala [hunk 1]
```diff
@@ -62,6 +62,11 @@ object NodeShapeModel extends AnyShapeModel {
                            Shapes + "dependencies",
                            ModelDoc(ModelVocabularies.Shapes, "dependencies", "Dependent properties constraint"))
 
+  val UnevaluatedProperties = Field(
+    ShapeModel,
+    Shapes + "unevaluatedProperties",
+    ModelDoc(ModelVocabularies.Shapes, "unevaluatedProperties", /* TODO: Add Something here*/ ""))
```
Comment (antoniogarrote) [inline amf-webapi/shared/src/main/scala/amf/plugins/domain/shapes/metamodel/NodeShapeModel.scala]: Missing documentation

File: amf-webapi/shared/src/main/scala/amf/plugins/domain/shapes/metamodel/ArrayShapeModel.scala [hunk 2]
```diff
@@ -33,12 +29,40 @@ class DataArrangementShape extends AnyShapeModel {
...
   val CollectionFormat = Field(
     Str,
     Shapes + "collectionFormat",
     ModelDoc(ModelVocabularies.Shapes, "collectionFormat", "Input collection format information"))
 
-  val specificFields               = List(Items, Contains, MinItems, MaxItems, UniqueItems, CollectionFormat)
+  val UnevaluatedItems = Field(
+    ShapeModel,
+    Shapes + "unevaluatedItems",
+    ModelDoc(ModelVocabularies.Shapes, "unevaluatedItems", /* TODO: Add Something here*/ ""))
```
Comment (antoniogarrote) [inline amf-webapi/shared/src/main/scala/amf/plugins/domain/shapes/metamodel/ArrayShapeModel.scala]: Missing description

File: documentation/model.md [hunk 3]
```diff
@@ -286,6 +286,9 @@ Shape that contains a nested collection of data shapes
  | maxCount | int | Maximum items count constraint | http://www.w3.org/ns/shacl#maxCount |
  | uniqueItems | boolean | Unique items constraint | http://a.ml/vocabularies/shapes#uniqueItems |
  | collectionFormat | string | Input collection format information | http://a.ml/vocabularies/shapes#collectionFormat |
+ | unevaluatedItems | [Shape](#shape) | Base class for all shapes. Shapes are Domain Entities that define constraints over parts of a data graph. They can be used to define and enforce schemas for the data graph information through SHACL. Shapes can be recursive and inherit from other shapes. | http://a.ml/vocabularies/shapes#unevaluatedItems |
```
Comment (antoniogarrote) [inline documentation/model.md]: Wrong description?

File: documentation/model.md [hunk 4]
```diff
@@ -925,6 +928,9 @@ Data shape containing nested multi-dimensional collection shapes
...
```
Comment (antoniogarrote) [inline documentation/model.md]: Wrong description?

File: documentation/model.md [hunk 5]
```diff
@@ -1132,6 +1138,7 @@ Shape that validates a record of fields, like a JS object
  | property | [[PropertyShape](#propertyshape)] | Constraint over a property in a data shape. | http://www.w3.org/ns/shacl#property |
  | propertyNames | [Shape](#shape) | Base class for all shapes. Shapes are Domain Entities that define constraints over parts of a data graph. They can be used to define and enforce schemas for the data graph information through SHACL. Shapes can be recursive and inherit from other shapes. | http://www.w3.org/ns/shacl#propertyNames |
  | dependencies | [[PropertyDependencies](#propertydependencies)] | Dependency between sets of property shapes | http://a.ml/vocabularies/shapes#dependencies |
+ | unevaluatedProperties | [Shape](#shape) | Base class for all shapes. Shapes are Domain Entities that define constraints over parts of a data graph. They can be used to define and enforce schemas for the data graph information through SHACL. Shapes can be recursive and inherit from other shapes. | http://a.ml/vocabularies/shapes#unevaluatedProperties |
```
Comment (antoniogarrote) [inline documentation/model.md]: wrong description?

File: documentation/model.md [hunk 6]
```diff
@@ -1402,6 +1409,7 @@ Dependency between sets of property shapes
  | ------ | ------ | ------ | ------ |
  | propertySource | url | Source property shape in the dependency | http://a.ml/vocabularies/shapes#propertySource |
  | propertyTarget | [url] | Target property shape in the dependency | http://a.ml/vocabularies/shapes#propertyTarget |
+ | propertyAppliedShape | [Shape](#shape) | Base class for all shapes. Shapes are Domain Entities that define constraints over parts of a data graph. They can be used to define and enforce schemas for the data graph information through SHACL. Shapes can be recursive and inherit from other shapes. | http://a.ml/vocabularies/shapes#propertyAppliedShape |
```
Comment (antoniogarrote) [inline documentation/model.md]: wrong description?

File: documentation/model.md [hunk 7]
```diff
@@ -1597,6 +1605,9 @@ Data shape describing a scalar value in the input data model, reified as an scal
  | Name | Value | Documentation | Namespace |
  | ------ | ------ | ------ | ------ |
  | datatype | url | Scalar range constraining this scalar shape | http://www.w3.org/ns/shacl#datatype |
+ | encoding | string | Describes the contents' value encoding | http://a.ml/vocabularies/shapes#encoding |
+ | mediaType | string | Describes the content's value mediatype | http://a.ml/vocabularies/shapes#mediaType |
+ | contentSchema | [Shape](#shape) | Base class for all shapes. Shapes are Domain Entities that define constraints over parts of a data graph. They can be used to define and enforce schemas for the data graph information through SHACL. Shapes can be recursive and inherit from other shapes. | http://a.ml/vocabularies/shapes#contentSchema |
```
Comment (antoniogarrote) [inline documentation/model.md]: wrong description?

**Review by antoniogarrote — CHANGES_REQUESTED** [review]
Some minor points about missing documentation

File: documentation/model.md (hunk 7 above)
Comment (tomsfernandez) [reply documentation/model.md]: fixed model exporter

File: amf-webapi/shared/src/main/scala/amf/plugins/domain/shapes/metamodel/ArrayShapeModel.scala (hunk 2 above)
Comment (tomsfernandez) [reply amf-webapi/shared/src/main/scala/amf/plugins/domain/shapes/metamodel/ArrayShapeModel.scala]: Filled description

File: amf-webapi/shared/src/main/scala/amf/plugins/domain/shapes/metamodel/NodeShapeModel.scala (hunk 1 above)
Comment (tomsfernandez) [reply amf-webapi/shared/src/main/scala/amf/plugins/domain/shapes/metamodel/NodeShapeModel.scala]: Filled description

**Review by antoniogarrote — DISMISSED** [review]
LGTM!

**Review by gutee — APPROVED** [review]

**Pull request merged**